.grit/patterns/go/channel_guarded_with_mutex.md

```diff
@@ -0,0 +1,180 @@
+---
+title: Remove the mutex. Channel guarded with mutex
+---
+
+Detected a channel guarded with a `mutex`. Channels already have an internal `mutex`, so this is unnecessary. Remove the mutex
+
+### references
+
+- [go-antipatterns](https://hackmysql.com/golang/go-antipatterns/#guarded-channel)
+
+tags: #fix #best-practice
+
+```grit
+language go
+
+`{ $body } ` where {
+    if( $body <: contains `$channel := make(chan $dataType)` ){
+        $body <: contains `var $mutax sync.Mutex` => .,
+        $body <: contains `$mutax.Lock()` => .,
+        $body <: contains `$mutax.Unlock()` => .,
+    }
+}
```

Comment:
This isn't a good use of if. `if` puts a *condition* on the predicts inside. The result would be completely unnecessary matches.

```suggestion
`{ $body } ` where {
    $body <: contains `$channel := make(chan $dataType)`)
    $body <: contains `var $mutax sync.Mutex` => .,
    $body <: contains `$mutax.Lock()` => .,
    $body <: contains `$mutax.Unlock()` => .,
}
```

Reply:
🤔, Alright, I will keep this mind for coming patterns